testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js

```diff
@@ -3960,6 +4029,75 @@ QUnit.module('searchEnabled', moduleSetup, () => {
 
         assert.strictEqual(loadStub.callCount, 3);
     });
+
+    QUnit.test('TagBox should add all clicked items after search if dataSource is async (T958611)', function(assert) {
+        const $tagBox = $('#tagBox').dxTagBox({
+            dataSource: getDSWithAsyncSearch(),
+            valueExpr: 'id',
+            displayExpr: 'id',
+            showSelectionControls: true,
+            searchEnabled: true,
+            searchExpr: 'id',
+            searchTimeout: TIME_TO_WAIT,
+            opened: true
+        });
+        const tagBox = $('#tagBox').dxTagBox('instance');
+
+        this.clock.tick(TIME_TO_WAIT * 3);
+        $('.dx-list-item').eq(0).trigger('dxclick');
+        this.clock.tick(TIME_TO_WAIT * 3);
+
+        const $input = $tagBox.find(`.${TEXTBOX_CLASS}`);
+        keyboardMock($input).type('search');
+
+        this.clock.tick(TIME_TO_WAIT * 4);
+        $('.dx-list-item').eq(0).trigger('dxclick');
```

Comment:
Maybe it's better to add `getListItems` function to module setup and call it to save items to `$listItems` variable before click to avoid copypaste?